src/orion/algo/robo/ablr/ablr.py

```diff
@@ -0,0 +1,113 @@
+""" Re-implementation of the ABLR model from [1].
+
+[1] [Scalable HyperParameter Transfer Learning](
+    https://papers.nips.cc/paper/7917-scalable-hyperparameter-transfer-learning)
+
+"""
+from __future__ import annotations
+
+from logging import getLogger as get_logger
+from typing import Sequence
+
+import torch
+from orion.algo.space import Space
+
+from orion.algo.robo.ablr.ablr_model import ABLR
+from orion.algo.robo.base import AcquisitionFnName, MaximizerName, RoBO
+
+logger = get_logger(__file__)
+
+
+# pylint: disable=unsubscriptable-object
+class RoBO_ABLR(RoBO[ABLR]):
+    """Implements the ABLR[1] algorithm, using RoBO.
+
+    [1] [Scalable HyperParameter Transfer Learning](
+        https://papers.nips.cc/paper/7917-scalable-hyperparameter-transfer-learning)
+
+
+    Parameters
+    ----------
+    space: ``orion.algo.space.Space``
+        Optimisation space with priors for each dimension.
+    seed: None, int or sequence of int
+        Seed to sample initial points and candidates points.
+        Default: 0.
+    n_initial_points: int
+        Number of initial points randomly sampled. If new points
+        are requested and less than `n_initial_points` are observed,
+        the next points will also be sampled randomly instead of being
+        sampled from the parzen estimators.
+        Default: ``20``
+    maximizer: str
+        The optimizer for the acquisition function.
+        Can be one of ``{"random", "scipy", "differential_evolution"}``.
+        Defaults to 'random'
+    acquisition_func: str
+        Name of the acquisition function. Can be one of ``['ei', 'log_ei', 'pi', 'lcb']``.
+    hparams: dict | ABLR.HParams | None
+        Hyperparameters for the ABLR model.
+        If None, the default hyperparameters will be used.
+    normalize_inputs: bool
+        Whether to normalize the inputs.
+    """
+
+    def __init__(
+        self,
+        space: Space,
+        seed: int | Sequence[int] | None = 0,
+        n_initial_points: int = 20,
+        maximizer: MaximizerName = "random",
+        acquisition_func: AcquisitionFnName = "ei",
+        hparams: ABLR.HParams | dict | None = None,
+        normalize_inputs: bool = True,
+    ):
+        super().__init__(
+            space=space,
+            seed=seed,
+            n_initial_points=n_initial_points,
+            maximizer=maximizer,
+            acquisition_func=acquisition_func,
+        )
+        self.hparams = hparams
+        self.normalize_inputs = normalize_inputs
+
+    # pylint: disable=missing-function-docstring
+    def build_model(self) -> ABLR:
+        return ABLR(
+            self.space,
+            hparams=self.hparams,
+            normalize_inputs=self.normalize_inputs,
+        )
+
+    # pylint: disable=missing-function-docstring
+    def seed_rng(self, seed: int) -> None:
+        # Seed the random + numpy + RoBo-specific RNG via the base class.
```

Comment:
```suggestion
    def seed_rng(self, seed: int) -> None:
        """Seed the random + numpy + RoBo-specific RNG via the base class."""
```
:man_shrugging: :)

Reply:
In general I don't add a docstring if the base class already has one. Otherwise I end up duplicating (at least part of) the docstring.

Reply:
Sounds reasonable especially if the docstring of the base class is reused in the generated doc. In this specific case however the fact that it is worth writing a comment to explain what it does hints that it may be worth making it a docstring.

Reply:
Fixed in 0ca473ee0fe70051acd0a43498770b2a5ef0bb2a